commands/load_balancers.go

```diff
@@ -378,12 +433,29 @@ func extractForwardingRules(s string) (forwardingRules []godo.ForwardingRule, er
 	return forwardingRules, err
 }
 
-func fillStructFromStringSliceArgs(obj any, s string) error {
+func extractDomains(s []string) (domains []*godo.LBDomain, err error) {
+	if len(s) == 0 {
+		return domains, err
+	}
+
+	for _, v := range s {
+		domain := new(godo.LBDomain)
+		if err := fillStructFromStringSliceArgs(domain, v, " "); err != nil {
```

Comment:
Sorry for all the back and forth here, but I did have one more question.

If feels odd to break the pattern used by other flags for `domains`. `glb-settings` takes a comma-separated list of key-value pairs as does `forwarding-rules`.

```
--domains "name:test-domain-1 is_managed:true certificate_id:test-cert-id-1"
--glb-settings target_protocol:http,target_port:80
--forwarding-rules entry_protocol:tcp,entry_port:3306,target_protocol:https,target_port:443
```

Can we stick to the existing pattern? I think it will be a more consistent experience for users.

Reply:
👋 The `glb-settings` are _flat_ config params which could be delimited by commas and we're parsing as expected. The `forwarding-rules` are a _slice_ of config rules: we actually have a limitation to how we handle this parameter in `doctl` currently. If you try passing multiple forwarding rules for an LB (as normally expected), only the latest rule will be used. E.g.:
```
./doctl \
...
--forwarding-rules "entry_protocol:http,entry_port:80,target_protocol:http,target_port:80" \
--forwarding-rules "entry_protocol:tcp,entry_port:50002,target_protocol:tcp,target_port:50002"
```

Running the above we're _only_ able to set the last forwarding rule specified (TCP). With `domains` the expectation should be the same, it takes a _slice_ of params and this adjustment to using a space delimiter makes it possible to parse multiple elements for a slice param appropriately.

Reply:
So basically TLDR:

- `forwarding-rules` accepts arg through `AddStringFlag` thus parses the param as a flat string, which isn't correct considering customers will have multiple forwarding rules
- `domains` accepts arg through `AddStringSliceFlag` thus parses the param as a slice of string, which is logically correct, except comma separator cannot be used to distinguish components in the same element